cpp/tests/test_abm_world.cpp

```diff
@@ -305,20 +306,45 @@ TEST(TestWorld, evolveMigration)
         auto world       = mio::abm::World(params);
         auto home_id     = world.add_location(mio::abm::LocationType::Home);
         auto work_id     = world.add_location(mio::abm::LocationType::Work);
+        auto hospital_id = world.add_location(mio::abm::LocationType::Hospital);
         auto icu_id      = world.add_location(mio::abm::LocationType::ICU);
 
         auto& p_dead = add_test_person(world, icu_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t);
+        auto& p_critical = add_test_person(world, hospital_id, mio::abm::AgeGroup::Age60to79,
+                                           mio::abm::InfectionState::InfectedCritical, t);
+        p_dead.set_assigned_location(icu_id);
         p_dead.set_assigned_location(home_id);
         p_dead.set_assigned_location(work_id);
+        p_critical.set_assigned_location(hospital_id);
+        p_critical.set_assigned_location(icu_id);
+
+        p_critical.add_new_infection(mio::abm::Infection(mio::abm::VirusVariant::Wildtype,
+                                                         mio::abm::AgeGroup::Age60to79, params, t + dt,
+                                                         mio::abm::InfectionState::Dead));
 
         mio::abm::TripList& data = world.get_trip_list();
         mio::abm::Trip trip1(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(9), work_id, home_id);
         mio::abm::Trip trip2(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(10), home_id, icu_id);
+        mio::abm::Trip trip3(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(10), home_id, icu_id);
+        mio::abm::Trip trip4(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(20), work_id, home_id);
+        mio::abm::Trip trip5(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(21), home_id, icu_id);
+        mio::abm::Trip trip6(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(21), home_id, icu_id);
```

Comment:
Why do you have doubled trips here? Did you mean to add trips for the p_severe (which is p_critical named at the moment)?

Reply:
These trips test if a person can still move (as scheduled) to other location than Cemetery after it's dead. Is it a good way to do it?

Reply:
That is a good test. I would just constrain it to a couple of trips. I think the current version looks fine.